server/src/test/java/org/elasticsearch/common/compress/lz4/ESLZ4CompressorTests.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.common.compress.lz4;
+
+import net.jpountz.lz4.LZ4Compressor;
+
+import net.jpountz.lz4.LZ4Factory;
+
+import net.jpountz.lz4.LZ4FastDecompressor;
+
+import org.elasticsearch.test.ESTestCase;
+
+import java.nio.charset.StandardCharsets;
+
+public class ESLZ4CompressorTests extends ESTestCase {
+
+    public void testCompress() {
+        for (int i = 0; i < 15; ++i) {
+            int stringLengthMultiplier = randomFrom(5, 10, 20, 40, 80, 160, 320);
+
+            final String uncompressedString = randomRealisticUnicodeOfCodepointLength(stringLengthMultiplier * 1024);
+            byte[] uncompressed = uncompressedString.getBytes(StandardCharsets.UTF_8);
```

Comment:
I wonder if we should generate random byte arrays instead, I would think that it should potentially provoke slightly more edge cases? At least it seems odd to test against UTF-8 strings only.